extension/src/background/helpers/dataStorage.ts

```diff
@@ -126,3 +127,53 @@ export const migrateSorobanRpcUrlNetworkDetails = async () => {
 
   await localStore.setItem(NETWORKS_LIST_ID, migratedNetworkList);
 }
+
+// This migration migrates the storage for custom tokens IDs to be keyed by network
+/*
+  Possible Schemas - 
+  should all be Futurenet in first schema
+
+  before --
+  {
+    [KEY_ID]: tokenList,
+    ...
+  }
+
+  after --
+  {
+    [SorobanClient.Network]: {
+      [KEY_ID]: tokenList,
+      ...
+    },
+    ...
+  }
+*/
+export const migrateTokenIdList = async () => {
+  const localStore = dataStorageAccess(browserLocalStorage);
+  const tokenIdsByKey = await localStore.getItem(TOKEN_ID_LIST) as Record<string, object>
+
+  const isNetwork = (key: string) => (Object.keys(Networks) as Array<keyof typeof Networks>).some((network) => key === network)
+
+  let newStorageSchema = tokenIdsByKey
+  if (tokenIdsByKey) {
+    // intially this was assumed to be Futurenet
+    const futurenetNetworks = Object.keys(tokenIdsByKey).reduce((prev, curr) => {
+      const next = prev
+      // if the key is not a network, assume old schema and move all keys(key ids) under futurenet key
+      if (!isNetwork(curr)) {
```

Comment:
I believe this would be truthy in the new schema if a user added a token under a custom network. I believe curr would equal `CUSTOM` in this case

Reply:
I like your idea of adding a schema version! Would it be easier if we introduce a schema version in storage and say `if (!schemaVersion || schemaVersion < 5.5.0)`: don't do any migration?